include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -438,4 +467,45 @@ def TT_AssertOp : TT_Op<"assert", [MemoryEffects<[MemWrite]>]> {
   let assemblyFormat = "$condition `,` $message `,` $file `,` $func `,` $line attr-dict `:` type($condition)";
 }
 
+//
+// Make a Tile Pointer
+//
+def TT_MakeTilePtrOp : TT_Op<"make_tile_ptr",
+                             [Pure,
+                              SameVariadicOperandSize,
+                              TypesMatchWith<"infer pointer type from the result type",
+                                             "result", "base",
+                                             "getPointerType(getElementTypeOfTilePointerType($_self))">]> {
+  let summary = "Make a tile pointer type with meta information of the parent tensor and the tile";
+
+  let description = [{
+      `tt.make_tile_ptr` takes both meta information of the parent tensor and the tile and returns a pointer to a tile,
+      e.g. returns a type of `tt.ptr<tensor<8x8xf16>>`.
+  }];
+
+  let arguments = (ins
+    TT_Ptr:$base,
+    Variadic<I64>:$shape,
+    Variadic<I64>:$strides,
+    Variadic<I32>:$offsets,
+    DenseI32ArrayAttr:$order
+  );
+
+  let results = (outs TT_Ptr:$result);
+
+  // Add additional `[]` to increase readability and split variadic lists
+  let assemblyFormat = "$base `,` `[` $shape `]` `,` `[` $strides `]` `,` `[` $offsets `]` attr-dict `:` type($result)";
+
+  let builders = [
+    OpBuilder<(ins
+        "Value":$base,
+        "ValueRange":$shape,
+        "ValueRange":$strides,
+        "ValueRange":$offsets,
+        "ArrayRef<int64_t>":$tile_shape,
```

Comment:
Triton shapes are int32_t, let's be explicit about this

Reply:
Fixed

Reply:
To be really explicit, I think the following would be better:

"Value":$base,
"ValueRange":$parentShape,
"ValueRange":$parentStrides,
"ValueRange":$parentOffsets,
"ArrayRef<int64_t>":$shape,
        
wdyt?

Reply:
I think that's better, should I also change the Python API?
`def make_block_ptr(base: tensor, shape, strides, offsets, block_shape, order, _builder=None)` into `def make_block_ptr(base: tensor, parent_shape, parent_strides, parent_offsets, parent_order, block_shape, _builder=None)`?

And:
```
"Value":$base,
"ValueRange":$parentShape,
"ValueRange":$parentStrides,
"ValueRange":$parentOffsets,
"ArrayRef<int32_t>":$tensorShape, // To align with Python API, block <-> tensor
```